### PR TITLE
DBZ-3055 Capture table state when LogMiner session fails

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3055

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java [hunk 1]
```diff
@@ -604,15 +598,73 @@ private static BigInteger resolveNextChangeFromScnRange(String nextChangeValue)
...
+
+    /**
+     * Helper method that dumps the result set of an arbitrary SQL query to the connector's logs.
+     *
+     * @param connection the database connection
+     * @param query the query to execute
+     * @throws SQLException thrown if an exception occurs performing a SQL operation
+     */
+    private static void logQueryResults(Connection connection, String query) throws SQLException {
+        try (PreparedStatement s = connection.prepareStatement(query); ResultSet rs = s.executeQuery()) {
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:655]: Would it help to use one of the utility methods on `JdbcConnection`?

**Review by gunnarmorling — APPROVED** [review]
LGTM. One question inline, no blocker.

@jpechane, can you include this for the relase?

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 1 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java]: Adjusted the API to pass in `OracleConnection` so we could access those methods.  I'll add making sure we align to using that class primarily for connection operations as it relates to `LogMinerHelper` in [DBZ-2552](https://issues.redhat.com/browse/DBZ-2552).

**Comment by gunnarmorling** [comment]
LGTM. Leaving the merge to @jpechane, as I'm not totally sure where he
stands in terms of release progress. Thanks!

>

**Pull request merged**